### DISCUSSION TITLE
Using `ruff` in pipes (formatting stdin and writing to stdout)?

### DISCUSSION THREAD

**Discussion by marinegor** [astral-sh/ruff#13690] [General]
Hi everyone,

short question: how to make something like this work:

```bash
$ cat foo.py

f = lambda x:x+3
f(int(input())

$ cat foo.py | ruff
f = lambda x : x + 3
f(int(input())
```

Reasoning behind it that I am developing in a large codebase that doesn't have a formatter configured by default, but still enjoy using one. So I run black on the pieces of the code that I add, and in Neovim it's done via selecting a piece of code and running `:'<,'>!black - 2>/dev/null`, which formats the code under selection.

I wonder now, if there's an option to replicate that with `ruff`? I've tried working with `ruff format --stdin-filename`, but no luck just yet.

**Comment by charliermarsh** [comment]
Are you looking for something like:
```
❯ cat foo.py | ruff format -
import os

sys
```

**Reply by marinegor** [reply]
ah ok indeed, that one's even simpler. Thanks!

(btw, insanely cool to see a reply appearing faster that I'm typing my own solution)

**Reply by pelson** [reply]
Just added https://github.com/astral-sh/ruff/issues/20460, as I don't believe it is possible to pipe `ruff check --fix` through stdout currently.

**Comment by marinegor** [comment]
Ok literally I posted it, I got the solution that works for selection buffer:

`!ruff format --stdin-filename % -`

this would apply `ruff format` to the selected block of code, which is exactly what I need

**Comment by timsat** [comment]
is it intentionally left undocumented via `ruff format --help` as of version 0.14.13?

**Reply by ntBre** [reply]
Which option is undocumented?

**Reply by timsat** [reply]
`ruff format -`

**Reply by ntBre** [reply]
Huh, yeah I guess this isn't documented in our CLI for any of the sub-commands. I think it would make sense to mention that, maybe in part of the `Arguments` documentation:

```
Arguments:
  [FILES]...  List of files or directories to check [default: .]
```

**Reply by timsat** [reply]
that would be really nice!
